warehouse/templates/manage/account.html

```diff
@@ -597,11 +609,9 @@ <h2>{% trans %}Security history{% endtrans %}</h2>
       <strong>{% trans %}API token added{% endtrans %}</strong><br>
       <small>
         {% trans %}Token name:{% endtrans %} {{ event.additional.description }}<br>
-        {% if event.additional.caveats.permissions == "user" %}
-          {% trans %}Token scope: entire account{% endtrans %}
-        {% else %}
-          {% trans project_name=event.additional.caveats.permissions.projects[0] %}Token scope: Project {{ project_name }}{% endtrans %}
-        {% endif %}
+        {% for caveat in event.additional.caveats %}
```

Comment:
I would strongly recommend testing this locally. With an  event created prior to applying these changes, `event.additional.caveats` looks like `{'version': 1, 'permissions': {'projects': ['sampleproject']}}` and this ends up passing the strings `'version'` and `'permissions'` to `caveat_detail`, which fails:

```
Traceback (most recent call last):
  File "/opt/warehouse/lib/python3.9/site-packages/pyramid_debugtoolbar/panels/performance.py", line 61, in resource_timer_handler
    result = handler(request)
  File "/opt/warehouse/lib/python3.9/site-packages/pyramid_tm/__init__.py", line 178, in tm_tween
    raise exc from None
  File "/opt/warehouse/lib/python3.9/site-packages/pyramid_tm/__init__.py", line 145, in tm_tween
    response = handler(request)
  File "/opt/warehouse/lib/python3.9/site-packages/pyramid/tweens.py", line 43, in excview_tween
    response = _error_handler(request, exc)
  File "/opt/warehouse/lib/python3.9/site-packages/pyramid/tweens.py", line 17, in _error_handler
    reraise(*exc_info)
  File "/opt/warehouse/lib/python3.9/site-packages/pyramid/util.py", line 733, in reraise
    raise value
  File "/opt/warehouse/lib/python3.9/site-packages/pyramid/tweens.py", line 41, in excview_tween
    response = handler(request)
  File "/opt/warehouse/src/warehouse/cache/http.py", line 74, in conditional_http_tween
    response = handler(request)
  File "/opt/warehouse/src/warehouse/sanity.py", line 76, in sanity_tween_egress
    return unicode_redirects(handler(request))
  File "/opt/warehouse/lib/python3.9/site-packages/pyramid/router.py", line 143, in handle_request
    response = _call_view(
  File "/opt/warehouse/lib/python3.9/site-packages/sentry_sdk/integrations/pyramid.py", line 90, in sentry_patched_call_view
    return old_call_view(registry, request, *args, **kwargs)
  File "/opt/warehouse/lib/python3.9/site-packages/pyramid/view.py", line 674, in _call_view
    response = view_callable(context, request)
  File "/opt/warehouse/lib/python3.9/site-packages/pyramid/config/views.py", line 151, in __call__
    return view(context, request)
  File "/opt/warehouse/lib/python3.9/site-packages/pyramid/config/views.py", line 170, in attr_view
    return view(context, request)
  File "/opt/warehouse/lib/python3.9/site-packages/pyramid/config/views.py", line 196, in predicate_wrapper
    return view(context, request)
  File "/opt/warehouse/src/warehouse/cache/http.py", line 33, in wrapped
    return view(context, request)
  File "/opt/warehouse/src/warehouse/cache/http.py", line 33, in wrapped
    return view(context, request)
  File "/opt/warehouse/lib/python3.9/site-packages/pyramid/viewderivers.py", line 512, in csrf_view
    return view(context, request)
  File "/opt/warehouse/lib/python3.9/site-packages/pyramid/viewderivers.py", line 319, in secured_view
    return view(context, request)
  File "/opt/warehouse/src/warehouse/manage/__init__.py", line 54, in wrapped
    return view(context, request)
  File "/opt/warehouse/src/warehouse/metrics/views.py", line 34, in wrapper_view
    return view(context, request)
  File "/opt/warehouse/lib/python3.9/site-packages/pyramid/viewderivers.py", line 450, in rendered_view
    response = view_renderer.render_view(
  File "/opt/warehouse/lib/python3.9/site-packages/pyramid/renderers.py", line 443, in render_view
    return self.render_to_response(response, system, request=request)
  File "/opt/warehouse/lib/python3.9/site-packages/pyramid/renderers.py", line 466, in render_to_response
    result = self.render(value, system_values, request=request)
  File "/opt/warehouse/lib/python3.9/site-packages/pyramid/renderers.py", line 462, in render
    result = renderer(value, system_values)
  File "/opt/warehouse/lib/python3.9/site-packages/pyramid_jinja2/__init__.py", line 261, in __call__
    return template.render(system)
  File "/opt/warehouse/lib/python3.9/site-packages/jinja2/environment.py", line 1291, in render
    self.environment.handle_exception()
  File "/opt/warehouse/lib/python3.9/site-packages/jinja2/environment.py", line 925, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/opt/warehouse/src/warehouse/templates/manage/account.html", line 211, in top-level template code
    <a href="#modal-close" class="button button--primary modal__action">{% trans %}Close{% endtrans %}</a>
  File "/opt/warehouse/src/warehouse/templates/manage/manage_base.html", line 408, in top-level template code
    <li>{{ error }}</li>
  File "/opt/warehouse/src/warehouse/templates/base.html", line 138, in top-level template code
    {% block body %}
  File "/opt/warehouse/src/warehouse/templates/base.html", line 245, in block 'body'
    {% block content %}
  File "/opt/warehouse/src/warehouse/templates/manage/manage_base.html", line 237, in block 'content'
    {% block main %}{% endblock %}
  File "/opt/warehouse/src/warehouse/templates/manage/account.html", line 654, in block 'main'
    <td>{{ event_summary(event) }}</td>
  File "/opt/warehouse/lib/python3.9/site-packages/jinja2/runtime.py", line 828, in _invoke
    rv = self._func(*arguments)
  File "/opt/warehouse/src/warehouse/templates/manage/account.html", line 613, in template
    {{ caveat_detail(caveat) }}
  File "/opt/warehouse/lib/python3.9/site-packages/jinja2/runtime.py", line 828, in _invoke
    rv = self._func(*arguments)
  File "/opt/warehouse/src/warehouse/templates/manage/account.html", line 469, in template
    {% trans project_name=caveat.permissions.projects[0] %}Token scope: Project {{ project_name }}{% endtrans %}
  File "/opt/warehouse/lib/python3.9/site-packages/jinja2/environment.py", line 474, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'str object' has no attribute 'permissions'
```

Reply:
Thanks, will do (and sorry that you caught that instead of me!)

Reply:
[d35585d](https://github.com/pypa/warehouse/pull/11157/commits/d35585db647c58896df317a6ff545ae41b87074b) should address this -- I confirmed that both old and new-style events render correctly, for both user-scoped and project-scoped API tokens.